backend/src/main/kotlin/blue/mild/covid/vaxx/service/IsinRetryService.kt

```diff
@@ -0,0 +1,153 @@
+package blue.mild.covid.vaxx.service
+
+import blue.mild.covid.vaxx.dao.repository.PatientRepository
+import blue.mild.covid.vaxx.dto.internal.PatientValidationResult
+import blue.mild.covid.vaxx.dto.request.IsinJobDtoIn
+import blue.mild.covid.vaxx.dto.response.IsinJobDtoOut
+import blue.mild.covid.vaxx.dto.response.PatientDtoOut
+import blue.mild.covid.vaxx.dto.response.toPatientVaccinationDetailDto
+import mu.KLogging
+
+
+class IsinRetryService(
+    private val patientService: PatientService,
+    private val patientValidation: PatientValidationService,
+    private val isinService: IsinServiceInterface,
+    private val dataCorrectnessService: DataCorrectnessService,
+    private val patientRepository: PatientRepository,
+    private val vaccinationService: VaccinationService,
+) {
+
+    private companion object : KLogging()
+
+    @Suppress("ComplexCondition", "ComplexMethod") // this is complex job, it is ok here
+    suspend fun runIsinRetry(isinJobDto: IsinJobDtoIn): IsinJobDtoOut {
+        val stats = IsinJobDtoOut()
+
+        val patients = patientService.getPatientsByConjunctionOf(
+            n = isinJobDto.patientsCount,
+            offset = isinJobDto.patientsOffset.toLong()
+        )
+
+        for (patient in patients) {
+            logger.debug("Checking ISIN id of patient ${patient.id}")
+
+            // 1. If patient has personal number but ISIN id is not set -> try ISIN validation
+            val updatedPatient = if (!patient.isinId.isNullOrBlank()) {
+                patient
+            } else if (isinJobDto.validatePatients && !patient.personalNumber.isNullOrBlank() ) {
+                logger.info("Patient ${patient.id} has personal number but no ISIN id. Validating in ISIN...")
+
+                val newIsinPatientId = retryPatientValidation(patient)
+
+                if (newIsinPatientId != null) {
+                    stats.validatedPatientsSuccess++;
+                } else {
+                    stats.validatedPatientsErrors++;
+                }
+                patient.copy(isinId = newIsinPatientId)
```

Comment:
je uz tady to isin id v databazi, nebo ne? pokud ne, tak je treba to tam propsat co nejdrive ho mame

pak taky bych pridal  vic logovani co se deje, ze se to podarilo / nepodarilo pro jakeho pacienta a co a jak se prave ted deje -- jde o to, ze nebude prostor to debugovat, takze potrebujeme videt logy abychom pri nejakem fixu sli co nejvice na jisto